formspree/app.py

```diff
@@ -2,18 +2,31 @@
 from flask import g
 from flask.ext.sqlalchemy import SQLAlchemy
 from flask.ext.login import LoginManager, current_user
-import fakeredis, redis
+import redis
 import settings
 
+
 DB = SQLAlchemy()
-if settings.TESTING:
-    REDIS = fakeredis.FakeStrictRedis()
-else:
-    REDIS = redis.StrictRedis.from_url(settings.REDIS_URL)
+_redis = None
+
+
+def REDIS():
+    ''' function to acquire the global redis connection '''
+    global _redis
+    if not _redis:
+        _redis = get_redis()
+    return _redis
```

Comment:
The main reason is that the code doesn't look pythonic. The "private" `_redis`, a function with all uppercase, etc. 

When you set `redis` variable to some value you are not defining a global, you are defining an attribute of the module like any function declaration. Example:
```
client = new Client()
def make_magic():
  pass
```
Both can be imported:
```
from module import client
from module import make_magic
```
Just because it's a instance of something it doesn't mean it's a global.

---
Now for the matter of having the patch decorator on every block you can avoid that by having a base test case that replace your redis instance.

Please don't get me wrong, I used to do this a lot when I was coding PHP, I would code my real code for the test (not talking about testability), like my code would have all sorts of getters and tricks in order be testable, but Python is so flexible you can code your real code for what is supposed to do and make your tests do the heavy lifting.